src/Common/DNSPTRResolver.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace DB {
+    struct DNSPTRResolver {
```

Comment:
This interface seems to be useless (we have only one class implementing it and it's not clear if we will ever have more)

Reply:
I like having interfaces because they cut dependencies and hide all implementation details. If we ever decide to use a different library or even rename the concrete class, clients do not need to change. It's part of the dependency inversion principle in SOLID.

It's a small detail, tho. If you think it's better without it, fine for me. Just let me know

Reply:
Makes sense, but I prefer avoiding unnecessary java-style abstractions unless there's a good reason to use them. If we add another library then we will extract an interface and change all usages (fortunately, there's only one place where it can be used, I mean `DB::DNSResolver`). Currently this interface solves non-existent problem.